Common/Data/Consolidators/RangeConsolidator.cs

```diff
@@ -0,0 +1,162 @@
+        /// <param name="volumeSelector">Extracts the volume from a data instance. The default value is null which does
+        /// not aggregate volume per bar, except if the input is a TradeBar.</param>
+        public RangeConsolidator(
+            decimal range,
+            Func<IBaseData, decimal> selector,
+            Func<IBaseData, decimal> volumeSelector = null)
+            : base(selector, volumeSelector)
+        {
+            Range = range;
+            _firstTick = true;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RangeConsolidator" /> class.
+        /// </summary>
+        /// <param name="range">The Range interval sets the range in which the price moves, which in turn initiates the formation of a new bar.
+        /// One range equals to one minimum price change, where this last value is defined depending of the RangeBar's symbol</param>
+        /// <param name="selector">Extracts the value from a data instance to be formed into a <see cref="RangeBar"/>. The default
+        /// value is (x => x.Value) the <see cref="IBaseData.Value"/> property on <see cref="IBaseData"/></param>
+        /// <param name="volumeSelector">Extracts the volume from a data instance. The default value is null which does
+        /// not aggregate volume per bar.</param>
+        public RangeConsolidator(decimal range,
+            PyObject selector,
+            PyObject volumeSelector = null)
+            : base(selector, volumeSelector)
+        {
+            Range = range;
+            _firstTick = true;
+        }
+
+        /// <summary>
+        /// Updates the current RangeBar being created with the given data.
+        /// Additionally, if it's the case, it consolidates the current RangeBar
+        /// </summary>
+        /// <param name="time">Time of the given data</param>
+        /// <param name="currentValue">Value of the given data</param>
+        /// <param name="volume">Volume of the given data</param>
+        protected override void UpdateBar(DateTime time, decimal currentValue, decimal volume)
+        {
+            bool isRising;
+            if (currentValue > CurrentBar.High)
+            {
+                isRising = true;
+            }
+            else if (currentValue < CurrentBar.Low)
+            {
+                isRising = false;
+            }
+            else
+            {
+                return;
+            }
+
+            CurrentBar.Update(time, currentValue, volume);
+            while (CurrentBar.IsClosed)
+            {
+                OnDataConsolidated(CurrentBar);
+                CurrentBar = new RangeBar(CurrentBar.Symbol, CurrentBar.EndTime, RangeSize, isRising ? CurrentBar.High + _minimumPriceVariation : CurrentBar.Low - _minimumPriceVariation, 0);
+                CurrentBar.Update(time, currentValue, Math.Abs(CurrentBar.Low - currentValue) > RangeSize ? 0 : volume); // Intermediate/phantom RangeBar's have zero volume
```

Comment:
> Math.Abs(CurrentBar.Low - currentValue) > RangeSize ? 0 : volume

This seems fishy I think, why using the Low and not the High for example?

Reply:
Because they're the same since the bar has just been created one live above, therefore `High, Low` and `Close` properties have the same value as `Open`.